### PR TITLE
Add note about managed image or shared gallery image

### PR DESCRIPTION
**Pull request by espoelstra**
# Description
Adds notes and code snippet on outputting a managed image or to a shared image gallery instead of a VHD.

#### Related issue:
#3855 

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [x] Documentation is updated (if applicable)
- [x] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: docs/create-image-and-azure-resources.md [hunk 1]
```diff
@@ -69,6 +69,79 @@ The function automatically creates all required Azure resources and kicks off pa
 
 *Please, check synopsis of `GenerateResourcesAndImage` for details about non-mandatory parameters.*
 
+##### Alternate output types (Managed Imaged/Shared Image Gallery)
+
+If you would like to use a more modern output image type you can add a small snippet to your pipeline to output a Managed Image or Shared Gallery Image instead.
+
+```
```
Comment (catthehacker) [inline docs/create-image-and-azure-resources.md:76]:
```suggestion
```pwsh
```

File: docs/create-image-and-azure-resources.md [hunk 2]
```diff
@@ -69,6 +69,79 @@ The function automatically creates all required Azure resources and kicks off pa
...
+
+# Print the resulting JSON
+$JSON | ConvertTo-Json -Depth 8 | Out-Host
+# Write the JSON to a file
+$JSON | ConvertTo-Json -Depth 8 | Out-File $templatePath -Encoding Ascii -Force
+```
+
+For a Shared Gallery Image you need to have created the gallery and the initial Image Definition beforehand (defining Windows or Linux as the osType as appropriate).
+
+```
```
Comment (catthehacker) [inline docs/create-image-and-azure-resources.md:106]:
```suggestion
```pwsh
```

**Comment by roryza** [comment]
I like the bit of code to change the packer command in the JSON. I think it would ideally just be integrated into the main GenerateResourcesAndImage script as an option, which could then apply to any builder JSON.

If you use {{env `ARM_SUBSCRIPTION_ID`}} (has to be the same) , {{env `ARM_RESOURCE_GROUP`}} (at least as a default this could be the same param) and {{env `ARM_RESOURCE_LOCATION`}} (more than likely the same) in the JSON there's less to "customise" (or add as paramters to GenerateResourcesAndImage) and you basically just need gallery name, image name, image version.

I also think it might read a bit confusing in the docs to have this new segment, and then immediately followed by the section on CreateAzureVMFromPackerTemplate which isn't relevant if you're using Image Gallery/Managed Images.

**Comment by miketimofeev** [comment]
@espoelstra We are sorry, but we decided not to add this guide to our docs even though it can be useful for someone. The reason is the maintenance burden — if something changes in the Managed Images/Shared Image Gallery then the deployment process can be broken and we will have to invest time in investigating the root cause and fixing the docs. We might get back to provide support for such types of images when we switch the image generation to use packer hcl files because it will allow us to create multiple configuration files with different variables.
Thank you!

**Comment by brgsstm** [comment]
Just want to add here that I've been using this solution with success, however now that Ubuntu2204 (and I assume other ImageType moving forward) is using hcl2, the soultion is no longer viable.  Modifying and re-outputting hcl2 appears to only be possible in Go, which is way more involved than modifying JSON.

**Comment by brgsstm** [comment]
Just to add - the reason we need to make these changes are due to Azure Policy that MS support as best practice, in that assigning public IPs to VMs is a bad idea.  MS also push a move to Managed Disk's rather than using VHDs so it would be nice to see this tooling make those best practices available as options.

**Pull request closed without merging**